articles/101_build_tool.md

```diff
@@ -289,59 +291,142 @@ The following items are possible extension points to provide custom functionalit
+Therefore features to easily filter the packages which need to be build are eagerly awaited to improve the development process.
+
+For the last ROS 2 release *Ardent* the buildfarm [build.ros.org](http://build.ros2.org) only provides jobs to generate Debian packages.
+Neither *devel* jobs or *pull request* jobs are available nor is it supported to build a local *prerelease*.
+For the coming ROS 2 release *Bouncy* these job types should be available to support maintainers.
+
+In ROS 2 *Bouncy* the univeral build tool will be the only supported option and `ament_tools` will be archived.
+
+#### Necessary work
+
+For either option **A)** or **B)** the follow items would need to be addressed:
+
+- The jobs and scripts on *ci.ros2.org* need to be updated to invoke the universal build tool instead of `ament_tools`.
+- The `ros_buildfarm` package needs to be updated to invoke the universal build tool instead of `catkin_make_isolated`.
+  The ROS 2 buildfarm would use this modification for the upcoming ROS 2 *Bouncy* release.
+  The ROS 1 buildfarm could use the same modification in the future.
+
+For option **A)** the follow items would need to be addressed:
+
+- Support for setup files generated by `ament_cmake`.
+- Support additional packages types: plain Python packages, CMake packages without a manifest.
+- Support for Windows using `.bat` files.
+- Support for the package manifest format version 3.
+
+For option **B)** the follow items would need to be addressed.
+
+- Address user feedback when the tool is being used by a broader audience.
+
+### Future
+
+The long term goal is that the universal build tool will be used in ROS 1, in ROS 2 as well as other non-ROS projects.
+There is currently no time line when the tool will be used on the ROS 1 build or be recommended to ROS 1 users.
+This solely depends on the resources available for ROS 1.
+
+Beside that for both options there is follow up work beyond the immediate goals.
+The following enumerates a few of them but is by no means exhaustive:
+
+For option **A)** the follow items should be considered:
+
+- Support for Python packages using a `setup.cfg` file.
+- Support for `PowerShell` to work around length limitations for environment variable on Windows.
+- Support to pass package specific argument.
+- Remove support for the *devel space* concept in ROS 1.
+- Update code base to Python 3.5+.
+- Refactor code base to reduce coupling (e.g. separate [API](https://github.com/catkin/catkin_tools/blob/2cae17f8f32b0193384d2c7734afee1c60c4add2/catkin_tools/execution/controllers.py#L183-L205) for output handling).
+- Additional functionality to build Gazebo including its dependencies.
+
+For option **B)** the follow items should be considered:
 
-- Refactor the software architecture of the existing code base to support the flexibility sketched by the extension points listed above.
+- Support cross compilation.
```

Comment:
I remember from offline discussion that there was no feature to be added here but just a lack of testing.
I recommend splitting the "missing features" from the "untested features" as it gives a better idea of how much work is still needed.

As a data point I cross-compiled ardent following [these instructions](https://github.com/ros2-for-arm/ros2/wiki/ROS2-on-arm-architecture) with colcon and it built successfully. I didnt have a board around to do runtime testing though

Reply:
Thank you for trying it. I removed the bullet for now (3b1925a282d3bf128e875ab17713ac591ca82729) since there seems to be no problem. If it turns out that there is still a problem it can be addressed as a bug fix.